### ISSUE TITLE
Glass (again)

### ISSUE THREAD

**Issue by legoboyvdlp** [c172p-team/c172p-detailed#1022]
![image](https://user-images.githubusercontent.com/10434911/37676565-2e465c1c-2c70-11e8-87fd-436db1333172.png)

**Comment by gilbertohasnofb** [comment]
What's wrong with that? Just some reflection distorted by the glass, isn't it?

**Comment by legoboyvdlp** [comment]
I'm a little concerned about the jagged lines. What exactly is it reflecting there? But perhaps the new reflection cubemap might reduce the severity of those lines?

**Comment by gilbertohasnofb** [comment]
> What exactly is it reflecting there?

Some random image that is not a true reflection of the world outside.

> But perhaps the new reflection cubemap might reduce the severity of those lines?

I don't think that's worth it to be honest. @wlbragg?

**Comment by wlbragg** [comment]
I don't remember exactly where I pointed it out, but I am not totally happy with the reflection cubes we  are using, Problem is I don't know that we  an "easily" improve upon them. They are Blender generated and are reflecting what you see in Blenders Model view, not real life. Because of the way models were historically put together they also don't contain any of the instruments. So the reflections aren't really even close but a distant facsimile.
My suggestion is someone try to create new maps from in the sim VS in Blender. The issue there is the scale is not square. But if you could start the sim in the most square resolution you can, and basically crop the screen shot square, maybe we could achieve satisfactory results. I think they would look better with missing data (because of the dimensions and having to square crop) than full data cubes but using a facsimile of the rendered visual.
I could be wrong, but I think it would look more realistic.
As far as this glass distortion were seeing now, I don't know that that will change much because it is being viewed at such an extreme angle, I think.

**Comment by gilbertohasnofb** [comment]
> They are Blender generated and are reflecting what you see in Blenders Model view

Are you sure about that? I took a look at them and they seemed to be using FG textures and everything. There are a bunch of cube maps in that FGDATA folder, but all of them look equally bad. :cry:

**Comment by wlbragg** [comment]
No. you right @gilbertohasnofb I was looking at the backup colored maps, specifically the red one...
![red_nx](https://user-images.githubusercontent.com/11069680/37690666-7f55424e-2c7a-11e8-9dc5-a7fa1d687692.png)
However the ones were using do appear to be rendered from in the sim.
![reflection_nx](https://user-images.githubusercontent.com/11069680/37690676-9133f686-2c7a-11e8-9fb7-04dbb595a26a.png)

**Comment by wlbragg** [comment]
Although they do appear to be too dark in my opinion. I think we may be loosing some detail. Also why the gray windows VS white? This was the original grayscale version that wasn't proportionally correct.
![grayscale_nx](https://user-images.githubusercontent.com/11069680/37690876-ced2036a-2c7b-11e8-9d4d-c96e5a3e0496.png)

**Comment by wlbragg** [comment]
The first image are the old wrong proportioned gray scale versions..
![fgfs-screen-006](https://user-images.githubusercontent.com/11069680/37691045-1d9ecbbc-2c7d-11e8-9294-f9772e747921.png)
![fgfs-screen-007](https://user-images.githubusercontent.com/11069680/37691054-295a782a-2c7d-11e8-9438-37f9ded0d5e4.png)
These are what we are using now...
![fgfs-screen-004](https://user-images.githubusercontent.com/11069680/37691092-5f9ef550-2c7d-11e8-8b55-833a7b9daeac.png)
![fgfs-screen-005](https://user-images.githubusercontent.com/11069680/37691095-60fd2462-2c7d-11e8-8b1f-f81979c9b588.png)

**Comment by gilbertohasnofb** [comment]
@wlbragg I think the reason we are using grey scale cockpit maps is because we have many different liveries for the cockpit but a single map, and back in the day we decided that a BW map would fit better with all of them. I still think that's an all right decision as the reflection is rarely that noticeable (in the sense of being able to really see cockpit parts and other things in the reflected image)

**Comment by wlbragg** [comment]
@gilbertohasnofb 

>I think the reason we are using grey scale cockpit maps is

I'm not questioning that, I agree. 

Now my only "potential" issue is the gray scale maps we are using now. They have a gray (or darker) window area that I think should really be white? The area that is window is actually gray in color and I don't think that is right.

Also, as a whole they appear to be a little darker than the original "gray scale" ones we used to use. Those we replaced because they were distorted proportionally.

**Comment by gilbertohasnofb** [comment]
I see, I will take a look at that.

**Comment by gilbertohasnofb** [comment]
@wlbragg done, reflection maps should have a white background, could you please confirm if that improves the issue?

**Comment by legoboyvdlp** [comment]
@wlbragg 

![image](https://user-images.githubusercontent.com/10434911/37911164-3691db12-3107-11e8-870a-7a9c9c31eaa1.png)

Is that line meant to be half green and half white on the back window?

**Comment by legoboyvdlp** [comment]
Also, at the rudder:
![image](https://user-images.githubusercontent.com/10434911/37911236-6beaa35c-3107-11e8-9bab-9f4b7470ff53.png)

A bit of a mistmatch there

**Comment by gilbertohasnofb** [comment]
> Is that line meant to be half green and half white on the back window?

No, but that's a livery mistake. I will fix that.

**Comment by gilbertohasnofb** [comment]
@legoboyvdlp I fixed this in the opened damage model PR branch `refactor-damage-model`

**Comment by legoboyvdlp** [comment]
Further problem:
![image](https://user-images.githubusercontent.com/10434911/37984644-b4f34e36-31ee-11e8-8bba-a5d977c649d4.png)

**Comment by gilbertohasnofb** [comment]
Looks to me that the ice formation is mapped just frontally and so it looks terrible on the sides. 

@legoboyvdlp turn off that HUD and fly this beauty using its own instruments!! :rofl:

**Comment by onox** [comment]
We should totally ban the HUD :stuck_out_tongue_winking_eye:

**Comment by wlbragg** [comment]
Can we verify the ice is not working on any sides? Also be aware of the fact that the front windshield is one piece and so if the side of the front windshield has ice then the side is also included in the effect. This may have to do with the effects switch used to distinguished between the two different angles of glass. The more horizontal VS vertical glass.

We need to revisit this whole fog, rain, reflection effect and make sure we didn't inadvertently screw it up with the changing  of the windshield.

I would like to see some good imaged of ice on all the windows.

**Comment by gilbertohasnofb** [comment]
> We should totally ban the HUD stuck_out_tongue_winking_eye

Someone please make a PR for the main FG project removing that thing! :rofl:

**Comment by gilbertohasnofb** [comment]
@wlbragg Sorry for my ignorance, but how can I force the ice to appear on the glass?

But it does look the modifications to the glass made the effects a bit strange, the rain also looks distorted on the sides:

![fg](https://user-images.githubusercontent.com/5700795/38000884-ba7bddae-3221-11e8-9edf-616a08a2cdb7.jpg)

**Comment by wlbragg** [comment]
>But it does look the modifications to the glass made the effects a bit strange, the rain also looks distorted on the sides:

Yes, that is a fact and there is no good solution to correct that.
The reason is we have two modes to do the rain in the effect
~~~
<surface-mapping-scheme type="int">1</surface-mapping-scheme>
~~~
and
~~~
<surface-mapping-scheme type="int">2</surface-mapping-scheme>
~~~
Mode 1 is for more horizontal glass faces and mode 2 is for the vertical glass faces.
Because our front window is one object now we can only assign mode 1. We used to assign mode 2 to the side objects. We still assign mode 2 to the door and aback seat side windows.

By the way, I just pushed the Blend and .ac turning on some objects I had off while doing the livery work. I forgot to turn them back on.

As far as the ice is concerned, I'll have to experiment before I can suggest a method to easily see it.

**Comment by wlbragg** [comment]
@gilbertohasnofb 
I just started from KEQA, I had the aircraft stared up and sitting on the runway at startup. It was sitting there for a little while and the I ctrl+U up to maybe 10-12 thousand feet. I opened the heater slider all the way for a minute or two, then opened the air slider all the way and it started to show the ice pretty quickly. I then shut both sliders and paused the simulation.
![fgfs-screen-008](https://user-images.githubusercontent.com/11069680/38001907-fc9f0828-31f4-11e8-84e4-538367b64bbd.png)

As you can see it is all working but indeed the windshield side glass is not looking good because of it being only one object and having to be assigned mode 1. The only solution is to separate the sides to their own objects so we can assign mode 2 to them.

**Comment by gilbertohasnofb** [comment]
@wlbragg I see, thanks for the detailed post above.

> The only solution is to separate the sides to their own objects so we can assign mode 2 to them.

What would be the side effect of doing that? Would that reinstate the issue of the glass looking a bit weird in the cockpit, which is what we had before?

**Comment by gilbertohasnofb** [comment]
Meaning, would it look again like this? https://github.com/c172p-team/c172p-detailed/issues/1010

**Comment by wlbragg** [comment]
Yes!

**Comment by gilbertohasnofb** [comment]
Hmm, so we have to choose between pretty glass vs pretty effects. What would be your preference? I am really unsure myself. The effects do not look good, but those are visible on 1% of flights while the front glass is always there. On the other, the divided glass does not look that bad and one can argue that that's totally acceptable in order to have the effects working well. What would you pick?

**Comment by wlbragg** [comment]
Here it is as before
![fgfs-screen-009](https://user-images.githubusercontent.com/11069680/38002279-dc857b24-31f6-11e8-8063-63a559dff773.png)
![fgfs-screen-010](https://user-images.githubusercontent.com/11069680/38002283-de8410e8-31f6-11e8-9c45-fa9f0e4dc2e8.png)

**Comment by gilbertohasnofb** [comment]
The effect certainly looks better, but the glass does look a bit worse. We need a choice: what would you pick?

**Comment by wlbragg** [comment]
For completeness, here it is now.
![fgfs-screen-014](https://user-images.githubusercontent.com/11069680/38002341-436debd2-31f7-11e8-8fb2-a4cc2c5257c7.png)
![fgfs-screen-013](https://user-images.githubusercontent.com/11069680/38002338-4017c5ac-31f7-11e8-92e6-52287a9e9cb1.png)

I guess with the hard edge? I would like @onox and @legoboyvdlp to weight in!

**Comment by gilbertohasnofb** [comment]
Thanks for the images. Yes, the effects do look terrible right now and the hard edge is not that much worse than we have, so I am fine reverting the glass into separate pieces. Let's see what others say.

**Comment by dany93** [comment]
> We should totally ban the HUD

I know that people use it, at least on other aircraft (to see their controls or the heading). I sometimes happen to be glad to find it, would it be only to find the direction of my starting airport. Cheating, yes... But during tests, cheating often is useful (like Ctrl-U, etc...).
Anyway, it only appears if you ask for it. Otherwise I don't see how it can be annoying.

With the UFO, it is essential.

**Comment by legoboyvdlp** [comment]
I'm happy for the glass to be reverted if it solves the effects.

**Comment by gilbertohasnofb** [comment]
> Anyway, it only appears if you ask for it. Otherwise I don't see how it can be annoying.

**Comment by dany93** [comment]
>  There are many screenshots with the HUD out there and they break my heart

In this case, if you are as much an expert in image processing as in sound processing (and patient) that might be the solution  :wink:

Sometimes, they also are here to show the GPS coordinates.

>  I promise I won't sent the PR removing it from the source code

Thank you for your tolerance!   :smile:

I thought of sending an image, just for you, with the HUD and the sky alone, without one of these ugly dashboards, but I don't want to pollute the Internet. Only an idea...

**Comment by wkitty42** [comment]
On 03/28/2018 10:08 AM, dany93 wrote:
> I thought of sending an image, just for you, with the HUD and the sky alone,
> without one of these ugly dashboards, but I don't want to pollute the
> Internet. Only an idea...

you mean like flying around and hitting the 'c' key to toggle off the whole 
craft and cockpit dash stuff so it looks like you're just floating along in the 
air but with somewhat louder engine noise?  i've done that... i've even flown 
entire flights without the craft or cockpit showing at all :) :) :)

but yeah, please leave the HUD alone... i definitely use it for coordinates 
amongst other things... heck, i've even been thinking of how to have a HUD 
projected on the windscreen of some craft because moving the view is too much to 
deal with and especially on a ""small"" monitor that is not HD or even widescreen...

**Comment by gilbertohasnofb** [comment]
> In this case, if you are as much an expert in image processing as in sound processing (and patient) that might be the solution wink

@dany93 Challenge accepted. His is @legoboyvdlp's screenshot before:

![](https://user-images.githubusercontent.com/10434911/37676565-2e465c1c-2c70-11e8-87fd-436db1333172.png)

And here is after:

![image](https://user-images.githubusercontent.com/5700795/38038392-bd1c8460-32a2-11e8-8b92-e23abb5af667.png)

Much better isn't it? :rofl:

**Comment by gilbertohasnofb** [comment]
It took me nearly 30 minutes to retouch @legoboyvdlp's image above for the joke, so someone please post a "hahaha" so that I don't feel that bad, please. :wink:

**Comment by wlbragg** [comment]
I was expecting it the other way around, for you to only leave the hud and get rid of the ugly panels!

**Comment by gilbertohasnofb** [comment]
@wlbragg There we go;

![image](https://user-images.githubusercontent.com/5700795/38052957-a9a0bd84-32ca-11e8-8192-7a171f70c2bd.png)

And this one only took 30 seconds!

**Comment by wkitty42** [comment]
On 03/28/2018 11:12 AM, Gilberto Agostinho wrote:
> Much better isn't it? 🤣


hahaha


:)



-- 
  NOTE: No off-list assistance is given without prior approval.
        *Please keep mailing list traffic on the list unless*
        *a signed and pre-paid contract is in effect with us.*

**Comment by gilbertohasnofb** [comment]
Thanks guys :smile:

**Comment by legoboyvdlp** [comment]
Should this be closed? Or is it open pending regeneration of cubemaps?
